packages/@groupby/elements-cache-driver-plugin/test/unit/common/cache-driver-plugin.test.ts

```diff
@@ -0,0 +1,87 @@
+import { CACHE_REQUEST } from '@groupby/elements-events';
+import { expect, spy } from '../../utils';
+import CacheDriverPlugin from '../../../src/cache-driver-plugin';
+
+describe('CacheDriverPlugin', () => {
+  let cacheDriverPlugin;
+
+  beforeEach(() => {
+    cacheDriverPlugin = new CacheDriverPlugin();
+  });
+
+  describe('metadata getter', () => {
+    it('should have the name `cache_driver`', () => {
+      expect(cacheDriverPlugin.metadata.name).to.equal('cache_driver');
+    });
+
+    it('should specify `cache` and `dom_events` as dependencies', () => {
+      expect(cacheDriverPlugin.metadata.depends).to.have.members(['cache', 'dom_events']);
+    });
+  });
+
+  describe('register()', () => {
+    it('should save the plugin registry for future use', () => {
+      const registry: any = { a: 1, b: 2 };
+
+      cacheDriverPlugin.register(registry);
+
+      expect(cacheDriverPlugin.core).to.equal(registry);
+    });
+  });
+
+  describe('ready()', () => {
+    it(`should listen for ${CACHE_REQUEST}`, () => {
+      const registerListener = spy();
+      cacheDriverPlugin.core = { dom_events: { registerListener } };
+
+      cacheDriverPlugin.ready();
+
+      expect(registerListener).to.be.calledWith(CACHE_REQUEST, cacheDriverPlugin.handleRequest);
+    });
+  });
+
+  describe('handleRequest()', () => {
+    const name = 'key';
+    const group = 'grp';
+    const returnEvent = 'ret';
+    let dispatchEvent;
+    let data;
+    let cache;
+
+    beforeEach(() => {
+      dispatchEvent = spy();
+      data = { a: 'a' };
+      cache = new Map();
+      cacheDriverPlugin.core = {
+        cache,
+        dom_events: { dispatchEvent },
+      };
+    });
+
+    it('should dispatch the requested data from the cache', () => {
+      const req = { detail: { name, returnEvent, group } };
+      cache.set('key::grp', data);
+
+      cacheDriverPlugin.handleRequest(req);
+
+      expect(dispatchEvent).to.be.calledWith(returnEvent, { name, data, group });
+    });
+
+    it('should expand an undefined group to the empty string', () => {
+      const req = { detail: { name, returnEvent } };
+      cache.set('key::', data);
+
+      cacheDriverPlugin.handleRequest(req);
+
+      expect(dispatchEvent).to.be.calledWith(returnEvent, { name, data, group: undefined });
+    });
+
+    it('should dispatch an undefined payload if no data is cached', () => {
```

Comment:
I'm starting to question whether this is a good idea. In what case would we want to initialize a component with an empty/undefined payload? Taking this action makes me think that we're opening ourselves up to future bugs where the component becomes empty when we didn't want it to.

I don't believe it's causing any bugs at the moment, and this is a valid choice, but I am questioning it.

Reply:
The trick is that we're not initializing the component with an empty/undefined payload. It's the responsibility of the receiver to handle an empty cache response and other unexpected data. In our case, if we get an empty response in the autocomplete component, we default to an empty array for results.

That said, I originally didn't have this as a test case. The behaviour I specified in the initial set of test cases was that the plugin forwarded what the cache provided. It just happens that the cache responds with `undefined` when the key doesn't exist.